File: IPGVolume/IPGVolume.Api/Models/Database/ScheduledVolumeChange.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace IPGVolume.Api.Models.Database
{
    public class ScheduledVolumeChange
    {
        public int? Id { get; set; }
        public float Setpoint { get; set; }
        public string CreatorName { get; set; }
        public string ClientKey { get; set; }
        public bool IsRecurring { get; set; }
        public DateTime ActiveOn { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime? ExpiresOn { get; set; }
        public DateTime? CompletedOn { get; set; }

        public List<RecurringDaysActive> RecurringDaysActive { get; set; }
    }

    public partial class ScheduledVolumeChangeConfiguration : IEntityTypeConfiguration<ScheduledVolumeChange>
    {
        public void Configure(EntityTypeBuilder<ScheduledVolumeChange> entity)
        {
            entity.HasMany(i => i.RecurringDaysActive).WithOne(i => i.ScheduledVolumeChange).HasForeignKey(i => i.ScheduledVolumeChangeId);
        }
    }
}

File: IPGVolume/IPGVolume.Api/ng-volume/src/app/models/scheduled-volume-change.ts
import { RecurringDaysActive } from "./recurring-days-active";

export class ScheduledVolumeChange {
    id?: number;
    setpoint: number;
    creatorName: string;
    clientKey: string;
    isRecurring: boolean;
    activeOn: Date;
    createdOn: Date;
    expiresOn?: Date;
    completedOn?: Date;

    recurringDaysActive: RecurringDaysActive[];

    constructor(obj: any) {
        Object.keys(obj).forEach(key => {
            this[key] = obj[key];
        });

        this.activeOn = new Date(this.activeOn);
        this.createdOn = new Date(this.createdOn);
        
        if(this.expiresOn) {
            this.expiresOn = new Date(this.expiresOn);
        }
        
        if(this.completedOn) {
            this.completedOn = new Date(this.completedOn);
        }

        this.recurringDaysActive = this.recurringDaysActive.map(i => new RecurringDaysActive(i));
    }

    get nice_setpoint(): string {
        return `${this.setpoint * 100}%`;
    }
}